client/block.c

```diff
@@ -1165,11 +1151,14 @@ static void *sync_thread(void *arg)
 
 static void reset_callback(struct ldus_rbtree *node)
 {
-	struct block_internal *b = (struct block_internal *)node;
-	if(b->remark) {
-		xdag_free(b->remark);
+	struct block_internal *bi = (struct block_internal *)node;
+	for(struct block_backrefs* to_free = get_first_blockref(bi); to_free != NULL; to_free = to_free->next){
+		xdag_free(to_free);
```

Comment:
it is bad idea to firstly do `xdag_free(to_free)` and then `to_free = to_free->next`

Reply:
why?

Reply:
my bad